mountpoint-s3/tests/fuse_tests/fork_test.rs

```diff
@@ -481,3 +627,40 @@ fn unmount(mount_point: &Path) {
 
     panic!("failed to unmount");
 }
+
+#[cfg(all(feature = "sse_kms", not(feature = "s3express_tests")))]
+struct ChildProcess(std::process::Child);
+
+#[cfg(all(feature = "sse_kms", not(feature = "s3express_tests")))]
+impl Drop for ChildProcess {
+    fn drop(&mut self) {
+        if !self
```

Comment:
This may need some comments. Also, is there no way to avoid changing the signature of `wait_for_exit`?

Reply:
I was thinking of adding a `wait_for_exit_not_owning`, but was not able to come up with a good name, which rang a bell in me, that probably that is not a good idea :)

it could be something like this: 
```
fn wait_for_exit_not_owning(child: &mut Child) -> ExitStatus {}

fn wait_for_exitg(mut child: Child) -> ExitStatus {
    wait_for_exit(&mut child)
}
```

otherwise -- this `drop` method can not move out its `self.0`:
```
cannot move out of `self` which is behind a mutable reference
```

(I can remove the whole `ChildProcess` thing, but it has a convenient functionality of printing MP's logs on test failure)

Reply:
Another option is to use `Option<Child>` (sorry) for the field and `take()` it in `drop`.

Reply:
But the main question was: why are you killing the process? Other tests call `unmount` at the end, any reason for the difference? If so, it would need a comment.